### PR TITLE
I am a material gun

### PR DESCRIPTION
**Pull request by RebeccaStankus**
TEST PLAN
-In desktop, go to limitless. Click on the gun below the arrow on the Color Cannon sign to equip it.
- An overlay should appear with directions on how to fire or unequip. Fire the gun. A particle effect should show splattering as a ball flies forward from the gun. When the ball hits something there should be a brief particle effect on the surface where it hit. Both particle effects should be the same color as the ball.
- Now the back of the gun should show a color that should match the next ball and particle effects when the gun is shot again. Firing the gun at most entities will swap the color on the back of the gun with the mesh of the entity that is hit. This should work on some avatars, such as the wooden mannequin. 
- Press 'u' to unequip the gun. It should remain where you unequipped it for 30 seconds and then disappear.  Equip another, unequip it and reeuip. This time, it should not disappear after 30 seconds and should have a lifetime of "-1".
-Enter HMD and equip the gun by pressing the bottom trigger on your controller inside the blue wireframe sphere that appears when you place your hand near it. Fire it by pressing the top trigger. Unequip by pressing the bottom trigger.

### CODE REVIEW

File: DomainContent/Hub/Material Gun/materials/Green.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "materialVersion": 1,
+  "materials": {
+    "model": "hifi_pbr",
+    "albedoMap": "https://hifi-content.s3.amazonaws.com/liv/dev/Materials/Colors/green.png"
```
Comment (elisa-lj11) [inline DomainContent/Hub/Material Gun/materials/Green.json]: move this into the Spot asset S3

File: DomainContent/Hub/Material Gun/materials/Orange.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "materialVersion": 1,
+  "materials": {
+    "model": "hifi_pbr",
+    "albedoMap": "https://hifi-content.s3.amazonaws.com/liv/dev/Materials/Colors/orange.png"
```
Comment (elisa-lj11) [inline DomainContent/Hub/Material Gun/materials/Orange.json]: move this into the Spot asset S3

File: DomainContent/Hub/Material Gun/materials/Red.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "materialVersion": 1,
+  "materials": {
+    "model": "hifi_pbr",
+    "albedoMap": "https://hifi-content.s3.amazonaws.com/liv/dev/Materials/Colors/red.png"
```
Comment (elisa-lj11) [inline DomainContent/Hub/Material Gun/materials/Red.json]: move this into the Spot asset S3

File: DomainContent/Hub/Material Gun/materials/Violet.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "materialVersion": 1,
+  "materials": {
+    "model": "hifi_pbr",
+    "albedoMap": "https://hifi-content.s3.amazonaws.com/liv/dev/Materials/Colors/violet.png"
```
Comment (elisa-lj11) [inline DomainContent/Hub/Material Gun/materials/Violet.json]: move this into the Spot asset S3

File: DomainContent/Hub/Material Gun/materials/Yellow.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "materialVersion": 1,
+  "materials": {
+    "model": "hifi_pbr",
+    "albedoMap": "https://hifi-content.s3.amazonaws.com/liv/dev/Materials/Colors/yellow.png"
```
Comment (elisa-lj11) [inline DomainContent/Hub/Material Gun/materials/Yellow.json]: move this into the Spot asset S3

**Comment by elisa-lj11** [comment]
@RebeccaStankus ping me when this is ready for a new CR 😄

File: DomainContent/Hub/Material Gun/materialGunSpawnerServer.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                    z: 10
+                },
+                parentID: gun,
+                collisionless: 1,
+                dynamic: 0,
+                name: "Gun Particle Effect",
+                isEmitting: false,
+                lifespan: 2,
+                maxParticles: 500,
+                textures: Script.resolvePath("particles/Bokeh-Particle.png"),
```
[the remark this answers is not in the document]
Comment (misslivirose) [reply DomainContent/Hub/Material Gun/materialGunSpawnerServer.js]: These are okay for properties

**Comment by thoys** [comment]
Tested successfully internally (can be merged)

**Pull request merged**